core/templates/services/keyboard-shortcut.service.ts

```diff
@@ -61,24 +59,24 @@ export class KeyboardShortcutService {
   bindLibraryPageShortcuts() {
     Mousetrap.bind('/', function() {
       var searchBar = <HTMLElement>document.querySelector(
-        '.protractor-test-search-input');
+        '.oppia-search-bar-text-input');
       searchBar.focus();
       return false;
     });
 
     Mousetrap.bind('c', function() {
-      document.getElementById('categoryBar').focus();
-      return false;
+      var categoryBar = <HTMLElement>document.querySelector(
+        '.oppia-category-bar-dropdown-toggle');
+      categoryBar.focus();
     });
 
     Mousetrap.bind('s', function() {
       document.getElementById('skipToMainContentId').focus();
-      return false;
     });
   }
 
   setHref(href) {
```

Comment:
Why do we need this function as a part of a shortcut service? I would suggest removing it and use `windowRef` in the service/directive/test which need to set href.

Reply:
So the reason I create this function is that jasmine does not allow page reloads in the testing environment. I can't 
spyOn the command window.location.href, So I created a wrapper function around it and used spyOn   setHref(href) in the test file. I think 
windowRef.nativewindow.location.href would have a similar issue? So a wrapper function is necessary unless there is a way to fake a page reload.

Reply:
Maybe Angular migration team can help here? @srijanreddy98 Do you have any thoughts/suggestions on @jimbyo issue?

Reply:
is this resolved?

Reply:
Done I was able to remove the setHref() by instead spying on 

```
    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
      location: {
        href: ''
      }
    });
```
This does come with some tradeoffs from the previous approach, however.
```
    spyOn(keyboardShortcutService, 'setHref').and.callFake(function(href) {
      hrefValue = href;
    });
```
In the previous approach, I was able to create expect statements like.
`    expect(hrefValue).toEqual('/get-started');`

I'm not exactly sure how to create similar expect statements with the current approach. Is it fine to have unit tests that don't assert against any condition and instead rely on the E2E tests to handle testing the page redirection?

Reply:
maybe we can do this:

```
describe:
    mockWindow = {
      location: {
        href: ''
      }
    }
    
    beforeEach:
        spyOnProperty(windowRef, 'nativeWindow').and.returnValue(mockWindow);

    it:
        // Setup.
        mockWindowwindowRef = ''
        
        // Baseline verification.
        expect(windowRefwindowRef).toBe('')

        // Action.
        KeyboardShortcutService....

        // Endline verification.

        Check the spyFunction get called.
        Check the href value through windowRef.
```

or something around this?